docs/resources/aws_transit_gateway_attachment.md

```diff
@@ -0,0 +1,102 @@
+  hvn_id         = "main-hvn"
+  cloud_provider = "aws"
+  region         = "us-west-2"
+  cidr_block     = "172.25.16.0/20"
+}
+
+resource "aws_vpc" "example" {
+  cidr_block = "172.31.0.0/16"
+}
+
+resource "aws_ec2_transit_gateway" "example" {
+  tags = {
+    Name = "example-tgw"
+  }
+}
+
+resource "aws_ram_resource_share" "example" {
+  name                      = "example-resource-share"
+  allow_external_principals = true
+}
+
+resource "aws_ram_principal_association" "example" {
+  resource_share_arn = aws_ram_resource_share.example.arn
+  principal          = hcp_hvn.main.provider_account_id
+}
+
+resource "aws_ram_resource_association" "example" {
+  resource_share_arn = aws_ram_resource_share.example.arn
+  resource_arn       = aws_ec2_transit_gateway.example.arn
+}
+
+resource "hcp_aws_transit_gateway_attachment" "example" {
+  depends_on = [
+    aws_ram_principal_association.example,
+    aws_ram_resource_association.example,
+  ]
+
+  hvn_id                        = hcp_hvn.main.hvn_id
+  transit_gateway_attachment_id = "example-tgw-attachment"
+  transit_gateway_id            = aws_ec2_transit_gateway.example.id
+  resource_share_arn            = aws_ram_resource_share.example.arn
+  destination_cidrs             = [aws_vpc.example.cidr_block]
+}
+
+resource "aws_ec2_transit_gateway_vpc_attachment_accepter" "example" {
+  transit_gateway_attachment_id = hcp_aws_transit_gateway_attachment.example.provider_transit_gateway_attachment_id
+}
+```
+
+## Schema
+
+### Required
+
+- **destination_cidrs** (List of String) The list of associated CIDR ranges.
+- **hvn_id** (String) The ID of the HashiCorp Virtual Network (HVN).
+- **resource_share_arn** (String, Sensitive) The Amazon Resource Name (ARN) of the Resource Share that is needed to grant HCP access to the Transit gateway in AWS.
+- **transit_gateway_attachment_id** (String) The ID of the Transit gateway attachment.
+- **transit_gateway_id** (String) The ID of the Transit gateway in AWS.
+
+### Optional
+
+- **id** (String) The ID of this resource.
+- **timeouts** (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+### Read-only
+
+- **created_at** (String) The time that the Transit gateway attachment was created.
+- **expires_at** (String) The time after which the Transit gateway attachment will be considered expired if it hasn't transitioned into 'Accepted' or 'Active' state.
+- **organization_id** (String) The ID of the HCP organization where the Transit gateway attachment is located. Always matches the HVN's organization.
+- **project_id** (String) The ID of the HCP project where the Transit gateway attachment is located. Always matches the HVN's project.
+- **provider_transit_gateway_attachment_id** (String) The Transit gateway attachment ID used by AWS.
+- **state** (String) The state of the Transit gateway attachment.
+
+<a id="nestedblock--timeouts"></a>
+### Nested Schema for `timeouts`
+
+Optional:
+
+- **create** (String)
```

Comment:
These fields, and the `default` field in the data source, are meant for setting custom timeouts for the operations. They render [like this](https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/resources/hvn#nested-schema-for-timeouts) on the docs site, and are currently documented this way for all of our resources due to `go generate`.

Similar to my comment on the `id`, I think we should leave these they way they are for consistency, but make some sort of adjustment moving forward so that they are more clear to users. Possible options include removing these blocks from the docs, or adding actual descriptions to explain what these fields do.